yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -262,6 +266,50 @@ private[spark] class ApplicationMaster(
 
   private def runDriver(securityMgr: SecurityManager): Unit = {
     addAmIpFilter()
+    if (isClusterMode) {
+      val containerId = ConverterUtils.toContainerId(
+        System.getenv(ApplicationConstants.Environment.CONTAINER_ID.name()))
+      val yarnClient = YarnClient.createYarnClient()
+      yarnClient.init(yarnConf)
+      yarnClient.start()
+      val addresses =
```

Comment:
The intent here is to ensure that we compare all possible hostnames for this node against the one that is in the NodeReport. It is possible there are multiple interfaces with different hostnames, so better be safe.